dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationHelper.java

```diff
@@ -0,0 +1,330 @@
+            }
+        }
+    }
+
+    /**
+     * This method allows deleting a single secret/property from a stored integration.
+     * @param form Secret specific-form
+     * @param user Logged in user.
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    void deleteSecret(final SecretForm form, final User user)
+            throws DotSecurityException, DotDataException {
+
+        final String serviceKey = form.getServiceKey();
+        if (!UtilMethods.isSet(serviceKey)) {
+            throw new DotDataException("Required param serviceKey isn't set.");
+        }
+        final String hostId = form.getHostId();
+        if (!UtilMethods.isSet(hostId)) {
+            throw new DotDataException("Required Param siteId isn't set.");
+        }
+        final Host host = hostAPI.find(hostId, user, false);
+        if(null == host) {
+            throw new DotDataException(String.format(" Couldn't find any host with identifier `%s` ",hostId));
+        }
+        final Optional<ServiceDescriptor> optionalServiceDescriptor = serviceIntegrationAPI
+                .getServiceDescriptor(serviceKey, user);
+        if (!optionalServiceDescriptor.isPresent()) {
+            throw new DotDataException(String.format("Unable to find a service descriptor bound to the  serviceKey `%s`. You must upload a yml descriptor.",serviceKey));
+        }
+        final Map<String, Param> params = form.getParams();
+        if(!UtilMethods.isSet(params)){
+            throw new DotDataException("Required Params aren't set.");
+        }
+        final ServiceDescriptor serviceDescriptor = optionalServiceDescriptor.get();
+        validateIncomingParamNames(params.keySet(),serviceDescriptor);
+
+        final Optional<ServiceSecrets> serviceSecretsOptional = serviceIntegrationAPI
+                .getSecretsForService(serviceKey, host, user);
+        if (!serviceSecretsOptional.isPresent()) {
+            throw new DotDataException(String.format("Unable to find a secret for serviceKey `%s`.",serviceKey));
+        } else {
+            serviceIntegrationAPI.deleteSecret(serviceKey, params.keySet(), host, user);
+        }
+    }
+
+    /**
+     * Validate the incoming params match the params described by a serviceDescriptor yml.
+     * @param paramNames a set of paramNames
+     * @param serviceDescriptor the service template
+     * @throws DotDataException This will give bac an exception if you send an invalid param.
+     */
+    private void validateIncomingParamNames(final Set<String> paramNames, final ServiceDescriptor serviceDescriptor)
+            throws DotDataException {
+
+        if(serviceDescriptor.isAllowExtraParameters()){
+            //If this flag has been specified we can have whatever param name we want added.
+           return;
+        }
+        final Map<String, Param> serviceDescriptorParams = serviceDescriptor.getParams();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)